playbooks/robusta_playbooks/persistent_volume_actions.py

```diff
@@ -0,0 +1,170 @@
+    finding = Finding(
+        title=f"Persistent Volume content",
+        source=FindingSource.MANUAL,
+        aggregation_key=function_name,
+        finding_type=FindingType.REPORT,
+        failure=False,
+    )
+
+    # Get persistent volume data the object contains data related to PV like metadata etc
+    pv = event.get_persistentvolume()
+    pv_claimref = pv.spec.claimRef
+
+    try:
+
+        if pv_claimref != None:
+            # Do this if there is a PVC attached to PV
+            pvc_obj = PersistentVolumeClaim.readNamespacedPersistentVolumeClaim(
+                name=pv_claimref.name, namespace=pv_claimref.namespace).obj
+            pod_obj = get_pod_related_to_pvc(pvc_obj, pv)
+            if (pod_obj != None):
+                # Do this if a Pod is using PVC
+
+                pod = RobustaPod.find_pod(
+                    pod_obj.metadata.name, pod_obj.metadata.namespace)
+                volume_mount_name = None
+
+                # Find name of the mounted volume on pod
+                for volume in pod.spec.volumes:
+                    if volume.persistentVolumeClaim.claimName == pv_claimref.name:
+                        volume_mount_name = volume.name
+                        break
+
+                # Use name of the mounted volume to find the correct volume mount
+                container_found_flag = False
+                container_volume_mount = None
+                for container in pod.spec.containers:
+                    if (container_found_flag):
+                        break
+                    if(container.volumeMounts):
+                        for volume_mount in container.volumeMounts:
+                            if volume_mount_name == volume_mount.name:
+                                container_volume_mount = volume_mount
+                                container_found_flag = True
+                                break
+                
+                result = pod.exec(
+                    f"ls -R {container_volume_mount.mountPath}/")
+                finding.title = f"Files present on persistent volume {pv.metadata.name} are: "
+                finding.add_enrichment(
+                    [
+
+                        FileBlock(
+                            f"Data.txt: ", result.encode()),
+
+                    ]
+                )
+            else:
+                # Do this if no Pod is attached to PVC
+                reader_pod = persistent_volume_reader(persistent_volume=pv)
+                result = reader_pod.exec(
+                    f"ls -R {reader_pod.spec.containers[0].volumeMounts[0].mountPath}")
+                finding.title = f"Files present on persistent volume {pv.metadata.name} are: "
+                finding.add_enrichment(
+                    [
+
+                        FileBlock(
+                            f"Data.txt: ", result.encode()),
+
+                    ]
+                )
+                # delete the reader pod
+                reader_pod.delete()
```

Comment:
I would put the delete pod, in a `finally` clause.
So if there's an exception, we'll still delete the reader pod, and not leave it in the cluster

Reply:
Resolved